crates/transaction-pool/src/blobstore/mod.rs

```diff
@@ -0,0 +1,105 @@
+//! Storage for blob data of EIP4844 transactions.
+
+#![allow(missing_docs)]
+
+pub use mem::InMemoryBlobStore;
+use reth_primitives::{bytes::Bytes, H256};
+use reth_rlp::{Decodable, RlpDecodable, RlpEncodable};
+use std::collections::BTreeMap;
+mod mem;
+
+// TODO replace with the actual type
+#[derive(Debug, Clone, RlpDecodable, RlpEncodable)]
+pub struct BlobSideCar;
+
+impl BlobSideCar {
+    pub fn size(&self) -> usize {
+        0
+    }
+}
+
+/// A blob store that can be used to store blob data of EIP4844 transactions.
+///
+/// This type is responsible for keeping track of blob data until it is no longer needed (after
+/// finalization).
+///
+/// Note: this is Clone because it is expected to be wrapped in an Arc.
+pub trait BlobStore: Send + Sync + Clone + 'static {
+    /// Inserts the blob sidecar into the store
+    fn insert(&self, tx: H256, data: BlobSideCar) -> Result<(), BlobStoreError>;
+
+    /// Inserts the blob sidecar into the store
+    fn insert_all(&self, txs: Vec<(H256, BlobSideCar)>) -> Result<(), BlobStoreError>;
+
+    /// Deletes the blob sidecar from the store
+    fn delete(&self, tx: H256) -> Result<(), BlobStoreError>;
+
+    /// Deletes all blob sidecars from the store
+    fn delete_all(&self, txs: Vec<H256>) -> Result<(), BlobStoreError>;
+
+    /// Retrieves the decoded blob data for the given transaction hash.
+    fn get(&self, tx: H256) -> Result<Option<BlobSideCar>, BlobStoreError> {
+        if let Some(raw) = self.get_raw(tx)? {
+            Ok(Some(BlobSideCar::decode(&mut raw.as_ref())?))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Retrieves all decoded blob data for the given transaction hashes
+    fn get_all(&self, txs: Vec<H256>) -> Result<Vec<(H256, BlobSideCar)>, BlobStoreError>;
```

Comment:
nit: should prob note that the number of returned items can be less than requested (if we don't have the blob stored)